app/src/main/java/lt/vilnius/tvarkau/mvp/interactors/ReportTypesInteractorImpl.kt

```diff
@@ -0,0 +1,34 @@
+package lt.vilnius.tvarkau.mvp.interactors
+
+import lt.vilnius.tvarkau.backend.LegacyApiService
+import lt.vilnius.tvarkau.backend.requests.GetReportTypesRequest
+import rx.Scheduler
+import rx.Single
+
+/**
+ * @author Martynas Jurkus
+ */
+class ReportTypesInteractorImpl(
+        val api: LegacyApiService,
+        val ioScheduler: Scheduler
+) : ReportTypesInteractor {
+
+    private val entryCache = mutableListOf<String>()
```

Comment:
By using immutable one you are guaranteeing that if something changes new list will be created and modifications will be performed. Also `entryCache::addAll` says that list can be modified and items added more than one time. Which is not true in this case.